src/lightning/fabric/plugins/collectives/collective.py

```diff
@@ -111,12 +110,12 @@ def destroy_group(cls, group: CollectibleGroup) -> None:
     def _convert_to_native_op(cls, op: str) -> Any:
         ...
 
-    def setup(self, **kwargs: Any) -> Self:
```

Comment:
Self is correct here

Reply:
OK and could it be also the class name?

Reply:
Self means the same object gets returned as the the method was called on. But "Collective" would mean any object of that class could be returned. So here "Self" is more accurate and therefore correct. I don't think this should be changed and "sphinx" is wrong here.

Also, why would sphinx care about types? We have mypy for this.

Reply:
it tries to link types to external docs so you can cross-link to the API in external docs
so good to revert to `Self` and set it as ignore class :)